config/configuration.go

```diff
@@ -177,10 +177,11 @@ func ValidateUrl(c *cli.Context, allowURLFromArgs bool) (*url.URL, error) {
 }
 
 type UnvalidatedIngressRule struct {
-	Hostname      string              `json:"hostname,omitempty"`
-	Path          string              `json:"path,omitempty"`
-	Service       string              `json:"service,omitempty"`
-	OriginRequest OriginRequestConfig `yaml:"originRequest" json:"originRequest"`
+	Hostname        string              `json:"hostname,omitempty"`
+	Path            string              `json:"path,omitempty"`
+	PathReplacement string              `yaml:"pathReplacement" json:"pathReplacement,omitempty"`
```

Comment:
Additionally, what are your thoughts on providing struct here that has two fields? Something like:
```go
type RewritePath struct {
    Regex string
    To    string
}
```
This should allow a bit more flexibility and readability in the Path regex versus the RewritePath regex for those that might want that.

Reply:
At the first thought, it's unnecessary because path is already a regex which could be safely reused for pattern matching, unlike in NGINX location where it has 3 variations (exact match, regex, prefix and suffix).
After thinking for a while, that might be better for cases where you have hundreds of similarly written ingresses with the same rewrite.

Current PR approach:
```yaml
  - hostname: subdomain.domain.com
    path: ^/serviceGroup/service/(.*)$
    rewrite: /$1
    service: https://localhost:3001
```

NGINX style approach:
```yaml
  - hostname: subdomain.domain.com
    path: ^/serviceGroup/service/
    rewrite:
      - regex: ^/\w+/\w+/(.*)$
        to: /$1
    service: https://localhost:3001
```
or
```yaml
  - hostname: subdomain.domain.com
    path: ^/serviceGroup/service/
    rewrite:
      - regex: ^/serviceGroup/service/(.*)$
        to: /$1
    service: https://localhost:3001
```
It is good because it follows NGINX style, however it allows space for unintentional bugs as regexes are duplicated, which means that they're not guaranteed to match the same path.
If you think that's fine - I'll change it to be this way, but I'd keep it as it is now.

Reply:
I think that makes sense, we can leave it as a string and it also sets a pretty good boundary. We don't want cloudflared to become a full reverse proxy with all the middleware, and extras. But having this simple rewrite on the path seems like a good middle-ground.